### PR TITLE
Fixes for flexible server in tutorial.

### PR DESCRIPTION
**Pull request by vmagelo**
## Purpose

We'll be updating tutorial https://learn.microsoft.com/en-us/azure/developer/python/tutorial-python-managed-identity-01 to used Flexible Server. And the user name for Flexible Server is different.

Also, the app_context is needed.

## Does this introduce a breaking change?

```
[x] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code and follow the tutorial linked above.

### CODE REVIEW

File: azureproject/production.py [hunk 1]
```diff
@@ -8,9 +8,9 @@
 CSRF_TRUSTED_ORIGINS = ['https://'+ os.environ['WEBSITE_HOSTNAME']] if 'WEBSITE_HOSTNAME' in os.environ else []
 
 # Configure Postgres database; the full username for PostgreSQL flexible server is
-# username (not @sever-name).
+# username (without @sever-name).
```
Comment (pamelafox) [inline azureproject/production.py:11]:
```suggestion
# username (without @server-name).
```

Might as well fix that typo if this line is changing

File: app.py [hunk 2]
```diff
@@ -39,8 +39,9 @@
 # Create databases, if databases exists doesn't issue create
 # For schema changes, run "flask db migrate"
 from models import Restaurant, Review
-db.create_all()
-db.session.commit()
+with app.app_context():
```
Comment (pamelafox) [inline app.py:42]: I had a different fix for the other Flask sample:
https://github.com/pamelafox/a-msdocs-flask-postgresql-sample-app-azd/commit/9f7cf103abb89093592be2888c600ceb46ff0ca1

In that repo, we expect the user to run `flask db upgrade`, so `db.create_all()` is not needed here. From what I understand, if we're using Flask-migrate, it's better practice to run `flask db upgrade`. (It also seems more production-level ready)

Comment (vmagelo) [reply app.py:42]: Ah, forgot that. Let me see I can use that. Thanks.

Comment (vmagelo) [reply app.py:42]: @pamelafox - I'm not having look remove those lines and following your changes.

Comment (vmagelo) [reply app.py:42]: Okay, think I figured it out and I can bring this app.py to be closer to what you have in the link above.

**Comment by pamelafox** [comment]
Looks nice! So are you going to store the migrations in the project folder, or have users generate the migrations? The other PG sample pre-makes the migrations, but it looks like from your azure-docs-pr that you'll ask users to generate them. I can't think of a reason right now to prefer one over the other.

**Review by pamelafox — APPROVED** [review]
Looks good from what I can see.

**Comment by vmagelo** [comment]
> Looks nice! So are you going to store the migrations in the project folder, or have users generate the migrations? The other PG sample pre-makes the migrations, but it looks like from your azure-docs-pr that you'll ask users to generate them. I can't think of a reason right now to prefer one over the other.

@pamelafox  - that's a good summary of what's being asked. For some reason I got in my head that having the /migrations folder checked in was bad. I can go back to what you have in https://github.com/pamelafox/a-msdocs-flask-postgresql-sample-app-azd and check in the migrations folder. I guess that has the advantage (once again like you keep prodding me to recognize  :-) ) that we would really only need "flask db upgrade" and nothing else. Let me try it out. Prob would be good to be consistent between samples.

**Comment by pamelafox** [comment]
I haven't personally used SQLAlchemy on a traffix-heavy production website but I believe most repos would check in the migrations directory (and a quick web search seemed to affirm that). If I was running a new migration, I'd probably check in the migrations in a staging branch, run upgrade with that branch, merge that branch, then deploy the code relying on the new migrations after. So the migrations would be checked in, just not at same time as new code relying on them, since it usually takes some time for database changes to be made on large tables.

**Review by pamelafox — APPROVED** [review]

**Comment by vmagelo** [comment]
Holding on this until we figure out form of new tutorial.

**Pull request merged**